### PR TITLE
Implemented check for variable rebinding

### PR DESCRIPTION
**Pull request by narrowtux**
This will check for multiple declaration of the same variable within a do block. 

For example,

```elixir
defmodule CredoSampleModule do
  def some_function() do
    a = 1
    a = 2
  end
end
```

would trigger a warning because `a` is declared twice.

### CODE REVIEW

File: test/credo/check/refactor/variable_rebind_test.exs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+""" |> to_source_file
+    |> assert_issue(@described_check)
+  end
+
+  test "it should report two violations" do
+"""
+defmodule CredoSampleModule do
+  def some_function() do
+    var_1 = 1 + 3
+    var_b = var1 + 7
```
Comment (rrrene) [inline test/credo/check/refactor/variable_rebind_test.exs]: 💩  greetings from hh.ex :-P

**Comment by narrowtux** [comment]
Proper tagging now: Implements #247

**Comment by narrowtux** [comment]
TODO: 
 -[ ] show last occurrence of same variable instead of first one
 -[ ] properly handle descructuring

**Comment by narrowtux** [comment]
@rrrene all done except the `@moduledoc`. 

Also I think that redeclaring should be an extra check because I feel like it should have a higher priority than this check.

**Comment by narrowtux** [comment]
Also don't want anyone to miss this gem:

image

**Comment by rrrene** [comment]
Thx! :+1:

**Pull request merged**